main.go

```diff
@@ -241,22 +209,112 @@ func main() {
 	if err := g.Run(); err != nil {
 		stdlog.Fatal(err)
 	}
+
+	level.Info(l).Log("msg", "up completed its mission!")
+}
+
+func registerMetrics(reg *prometheus.Registry) metrics {
+	m := metrics{
+		remoteWriteRequests: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "up_remote_writes_total",
+			Help: "Total number of remote write requests.",
+		}, []string{"result"}),
+		queryResponses: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "up_queries_total",
+			Help: "The total number of queries made.",
+		}, []string{"result"}),
+		metricValueDifference: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name:    "up_metric_value_difference",
+			Help:    "The time difference between the current timestamp and the timestamp in the metrics value.",
+			Buckets: prometheus.LinearBuckets(4, 0.25, 16),
+		}),
+	}
+	reg.MustRegister(
+		prometheus.NewGoCollector(),
+		prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}),
+		m.remoteWriteRequests,
+		m.queryResponses,
+		m.metricValueDifference,
+	)
+
+	return m
+}
+
+func parseOptions() (options, error) {
+	var (
+		rawEndpointWrite string
+		rawReadAddress   string
+	)
+
+	opts := options{}
+
+	flag.StringVar(&rawEndpointWrite, "endpoint-write", "", "The endpoint to which to make remote-write requests.")
+	flag.StringVar(&rawReadAddress, "read-address", "",
```

Comment:
these endpoint/address flags are inconsistent right now

Reply:
Yes, I'm working on those. The behaviour between that consumes those flags are different (what expects a host and appends path, the other expects fully-fledged endpoint). That's why I thought I should name them differently. However, it doesn't look neat, so I'm working on fixing this discrepancy.